cw_bitcoin/lib/electrum_wallet_addresses.dart

```diff
@@ -53,41 +51,71 @@ abstract class ElectrumWalletAddressesBase extends WalletAddresses with Store {
   @override
   @computed
   String get address {
-    if (receiveAddresses.isEmpty) {
-      final address = generateNewAddress().address;
-      return walletInfo.type == WalletType.bitcoinCash ? toCashAddr(address) : address;
-    }
-    final receiveAddress = receiveAddresses.first.address;
+    if (!isEnabledAutoGenerateSubaddress) {
+      if (receiveAddresses.isEmpty) {
+        final newAddress = generateNewAddress().address;
+        return walletInfo.type == WalletType.bitcoinCash ? toCashAddr(newAddress) : newAddress;
+      }
+      final receiveAddress = receiveAddresses.first.address;
 
-    return walletInfo.type == WalletType.bitcoinCash ? toCashAddr(receiveAddress) : receiveAddress;
+      return walletInfo.type == WalletType.bitcoinCash
+          ? toCashAddr(receiveAddress)
+          : receiveAddress;
+    } else {
+      final receiveAddress = (receiveAddresses.first.address != addresses.first.address &&
+              previousAddressRecord != null)
+          ? previousAddressRecord!.address
+          : addresses.first.address;
+
+      return walletInfo.type == WalletType.bitcoinCash
+          ? toCashAddr(receiveAddress)
+          : receiveAddress;
+    }
   }
 
+  @observable
+  bool isEnabledAutoGenerateSubaddress = true;
+
   @override
-  String get primaryAddress => getAddress(index: 0, hd: mainHd);
+  set address(String addr) {
+
+    String address;
+    if (addr.startsWith('bitcoincash:')) {
+      address = toLegacy(addr);
+    } else {
+      address = addr;
+    }
+    final addressRecord = addresses.firstWhere((addressRecord) => addressRecord.address == address);
+
+    previousAddressRecord = addressRecord;
+    receiveAddresses.remove(addressRecord);
+    receiveAddresses.insert(0, addressRecord);
```

Comment:
why do you remove it and then add it at the beginning?

Reply:
It updates the primary address for receiving funds in the wallet to the one selected by the user.